gitpod-io/gitpod
# [kots]: enable use of a local registry

## Description

This will enable use of Replicated's airgapped functionality.

## Related Issue(s)

Fixes #9180
Fixed #9041

## How to test

1. Use the [self-hosted infra repo](https://github.com/MrSimonEmms/gitpod-self-hosted-infrastructure) to create an Azure airgapped installation. Set the `TF_VAR_domain_name` envvar to the domain name you want to use and `TF_VAR_enabled_airgapped=true`.
2. Download the latest airgapped bundle on the `dev-sje` channel
3. Run `make install` to get the command to install Gitpod via KOTS. This will also configure the local registry
4. In [localhost:8800](http://localhost:8800), configure and install the application. **IMPORTANT** this will hide the Gitpod registry config - this is because we must use the same registry that images are pulled from.

Important, this has been tested with a valid TLS cert issued via LetsEncrypt. Self-signed/corporate CA certs are outside the scope of this ticket

This workflow doesn't include anything past login as that requires a self-hosted GitLab instance on the same network which I've not yet done. In theory, this should work fine but any further stuff for that can be done as a future PR.

## Release Notes

```release-note
[kots]: enable use of a local registry
```

## Documentation
